src/com/boxboat/jenkins/library/docker/Registry.groovy

```diff
@@ -26,8 +28,55 @@ class Registry extends BaseConfig<Registry> implements Serializable {
         return "${scheme}://${host}"
     }
 
-    def withCredentials(closure) {
-        Config.pipeline.docker.withRegistry(getRegistryUrl(), credential, closure)
+    void makeDockerConfig(String username, String password, String dir = "~/.docker") {
+        def fileName = "${dir}/config.json"
+        def reg = ["auths": [
+                "${getRegistryUrl()}": [
+                        "auth": "${username}:${password}".bytes.encodeBase64().toString()
+                ]]]
+
+        if (Config.pipeline.fileExists(fileName)) {
+            def data = Config.pipeline.readJSON(file: fileName)
+            reg['auths'] = data['auths'] + reg['auths']
+        }
+        Config.pipeline.writeJSON(file: fileName, json: reg)
     }
 
+    def withCredentials(Map configMap = [:], Closure closure) {
```

Comment:
I think this class needs the same treatment as the Vault Classes where it uses a `Params` objects with overloads, so that the original API definition remains

Also, can you use `DOCKER_CONFIG` to store the config in a temp directory that gets cleaned up after the call, example: https://github.com/boxboat/dockhand/blob/master/src/com/boxboat/jenkins/library/gcloud/GCloudRegistry.groovy#L12-L28

Reply:
Will update the definition.
The `DOCKER_CONFIG` is set to a temp directory, see line 52 `Utils.withTmpDir("DOCKER_CONFIG")`.  I can update the GCloudRegistry to be consistent with

Reply:
Ah ok, I think the default parameter in `makeDockerConfig` set to `String dir = "~/.docker"` was confusing me when reading through it

Reply:
I don't think you need to modify the GCloud stuff in this PR, it uses the gcloud command and doesn't call writeJSON so it can continue to use `mktemp`

Reply:
Added params and overloaded withCredentials

Updated all uses of `mktemp -d` to use `withTmpDir()`:
- src/com/boxboat/jenkins/library/gcloud/GCloudAccount.groovy
- src/com/boxboat/jenkins/library/gcloud/GCloudGKEDeployTarget.groovy
- src/com/boxboat/jenkins/library/gcloud/GCloudRegistry.groovy